arches_lingo/src/arches_lingo/components/generic/NoteEditor.vue

```diff
@@ -0,0 +1,243 @@
+const eventTypeOptions = ref<ControlledListItem[]>([]);
+const groupAndPersonOptions = ref<ResourceInstanceReference[]>();
+const textualWorkOptions = ref<ResourceInstanceReference[]>();
+
+function onUpdate(
+    node: keyof SchemeStatement,
+    val: ControlledListItem[] | ResourceInstanceReference[] | string,
+) {
+    if (Array.isArray(val)) {
+        (formValue.value[node] as unknown) = val.map((item) => toRaw(item));
+    } else {
+        (formValue.value[node] as unknown) = toRaw(val);
+    }
+}
+
+async function save() {
+    try {
+        await updateSchemeNote(
+            route.params.id as string,
+            formValue.value.tileid as string,
+            formValue.value,
+        );
+        emit("update");
+    } catch (error) {
+        toast.add({
+            severity: ERROR,
+            summary: $gettext("Error saving scheme"),
+            detail: (error as Error).message,
+        });
+    }
+}
+
+async function getControlledListOptions(
+    listId: string,
+): Promise<ControlledListItem[]> {
+    const parsed = await fetchControlledListOptions(listId);
+    const options = parsed.items.map(
+        (item: ControlledListItemResult): ControlledListItem => ({
+            list_id: item.list_id,
+            uri: item.uri,
+            labels: item.values,
+        }),
+    );
+    return options;
+}
+
+async function getResourceInstanceOptions(
+    fetchOptions: () => Promise<ResourceInstanceResult[]>,
+): Promise<ResourceInstanceReference[]> {
+    const options = await fetchOptions();
+    const results = options.map((option: ResourceInstanceResult) => {
+        const result: ResourceInstanceReference = {
+            display_value: option.descriptors[selectedLanguage.value.code].name,
+            resourceId: option.resourceinstanceid,
+            ontologyProperty: "ac41d9be-79db-4256-b368-2f4559cfbe55",
+            inverseOntologyProperty: "ac41d9be-79db-4256-b368-2f4559cfbe55",
+        };
+        return result;
+    });
+    return results;
+}
+
+onMounted(async () => {
+    const [languageOpts, typeOpts, statusOpts, metatypeOpts, eventTypeOpts] =
+        await Promise.all([
+            getControlledListOptions(LANGUAGE_CONTROLLED_LIST),
+            getControlledListOptions(LABEL_CONTROLLED_LIST),
+            getControlledListOptions(STATUSES_CONTROLLED_LIST),
+            getControlledListOptions(METATYPES_CONTROLLED_LIST),
+            getControlledListOptions(EVENT_TYPES_CONTROLLED_LIST),
+        ]);
+
+    languageOptions.value = languageOpts;
+    typeOptions.value = typeOpts;
+    statusOptions.value = statusOpts;
+    metatypeOptions.value = metatypeOpts;
+    eventTypeOptions.value = eventTypeOpts;
+    groupAndPersonOptions.value = await getResourceInstanceOptions(
+        fetchGroupRdmSystemList,
+    );
+    groupAndPersonOptions.value = [
+        ...(groupAndPersonOptions.value || []),
+        ...(await getResourceInstanceOptions(fetchPersonRdmSystemList)),
+    ];
```

Comment:
this could stand to be refactored

Reply:
yanked out select initialize into its own method.